### PR TITLE
PR for Crunchy PostgreSQL Cluster Helm Example

### PR DESCRIPTION
**Pull request by jmccormick2001**
here is the crunchy postgresql helm example, I tested it on Kube 1.4.5 and Helm 2.0.0.rc.1

### CODE REVIEW

File: incubator/crunchy-postgresql-cluster/templates/master-pod.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    "helm.sh/created": {{.Release.Time.Seconds | quote }}
+spec:
+  # This shows how to use a simple value. This will look for a passed-in value
+  # called restartPolicy. If it is not found, it will use the default value.
+  # {{default "Never" .restartPolicy}} is a slightly optimized version of the
+  # more conventional syntax: {{.restartPolicy | default "Never"}}
+  restartPolicy: {{default "Never" .Values.restartPolicy}}
+  containers:
+  - name: server
+    image: "crunchydata/crunchy-postgres:centos7-9.5-1.2.4"
```
Comment (lachie83) [inline incubator/crunchy-postgresql-cluster/templates/master-pod.yaml]: Please add these ass variables in values.yaml

File: incubator/crunchy-postgresql-cluster/templates/master-pod.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        memory: {{default "100M" .Values.Memory}}
+    env:
+    - name: PGHOST
+      value: /tmp
+    - name: PG_MASTER_USER
+      value: master
+    - name: PG_MODE
+      value: master
+    - name: PG_MASTER_PASSWORD
+      value: "{{.Values.MasterPassword}}"
```
Comment (lachie83) [inline incubator/crunchy-postgresql-cluster/templates/master-pod.yaml:41]: How about we put the passwords in a secret and then pull them into envvars?

File: incubator/crunchy-postgresql-cluster/templates/master-pod.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    - name: PG_ROOT_PASSWORD
+      value: "{{.Values.RootPassword}}"
+    volumeMounts:
+    - mountPath: /pgdata
+      name: pgdata
+    - mountPath: /backup
+      name: backup
+      readOnly: true
+  volumes:
+  - emptyDir: {}
```
Comment (lachie83) [inline incubator/crunchy-postgresql-cluster/templates/master-pod.yaml]: Please add support for PV if this is storing non-ephemeral content

File: incubator/crunchy-postgresql-cluster/templates/master-service.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+kind: Service
+metadata:
+  name: "{{.Values.MasterName}}"
+  labels:
+    name: "{{.Values.MasterName}}"
+    heritage: {{ .Release.Service | quote }}
+    release: {{ .Release.Name | quote }}
+    chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+spec:
+  type: ClusterIP
```
Comment (lachie83) [inline incubator/crunchy-postgresql-cluster/templates/master-service.yaml]: Should we convert the type to a variable

File: incubator/crunchy-postgresql-cluster/templates/master-pod.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{.Values.MasterName}}"
```
Comment (lachie83) [inline incubator/crunchy-postgresql-cluster/templates/master-pod.yaml]: I would suggest using a name that includes the release name to covert multiple installs in the same namespace

File: incubator/crunchy-postgresql-cluster/templates/master-pod.yaml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  annotations:
+    "helm.sh/created": {{.Release.Time.Seconds | quote }}
+spec:
+  # This shows how to use a simple value. This will look for a passed-in value
+  # called restartPolicy. If it is not found, it will use the default value.
+  # {{default "Never" .restartPolicy}} is a slightly optimized version of the
+  # more conventional syntax: {{.restartPolicy | default "Never"}}
+  restartPolicy: {{default "Never" .Values.restartPolicy}}
+  containers:
+  - name: server
```
Comment (lachie83) [inline incubator/crunchy-postgresql-cluster/templates/master-pod.yaml]: Should this be something from values.yaml?

**Review by lachie83 — CHANGES_REQUESTED** [review]
Thanks for the submission. I took a first pass at review. Please take a look at this document for tech requirements of charts. https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements

**Comment by lachie83** [comment]
@jmccormick2001 Can you please address the following -- "commit missing GitHub user"

**Comment by jmccormick2001** [comment]
added support for PVs

File: incubator/crunchy-postgresql-cluster/templates/master-pod.yaml (hunk 6 above)
Comment (jmccormick2001) [reply incubator/crunchy-postgresql-cluster/templates/master-pod.yaml]: made the container name a value

File: incubator/crunchy-postgresql-cluster/templates/master-pod.yaml (hunk 5 above)
Comment (jmccormick2001) [reply incubator/crunchy-postgresql-cluster/templates/master-pod.yaml]: included the release name into the name

File: incubator/crunchy-postgresql-cluster/templates/master-pod.yaml (hunk 2 above)
Comment (jmccormick2001) [reply incubator/crunchy-postgresql-cluster/templates/master-pod.yaml:41]: added secrets for holding passwords

Comment (jmccormick2001) [reply incubator/crunchy-postgresql-cluster/templates/master-pod.yaml:41]: done, added secrets to hold passwords.

**Review by jmccormick2001 — COMMENTED** [review]
I think I got all the changes made you had marked in your review.  cheers.

File: incubator/crunchy-postgresql-cluster/NOTES.txt [hunk 7]
```diff
@@ -0,0 +1,49 @@
+=== Testing
```
Comment (viglesiasce) [inline incubator/crunchy-postgresql-cluster/NOTES.txt]: The NOTES.txt file needs to be moved to the templates folder.

**Review by viglesiasce — CHANGES_REQUESTED** [review]

**Comment by jmccormick2001** [comment]
relocated the NOTES.txt, also added comments about NFS

**Comment by viglesiasce** [comment]
This chart does not work out of the box.
Can you either:
- Use a dynamic PVC like [the postgresql chart](https://github.com/kubernetes/charts/tree/master/stable/postgresql/templates)
- Add a dependent NFS server chart

**Comment by viglesiasce** [comment]
This would benefit from using a PetSet/StatefulSet of size 1. Currently, if the master pod is deleted it wont be recreated. With PetSet/StatefulSet you would get a persistent network identity and a PVC.

**Comment by jmccormick2001** [comment]
the use of StatefulSet is an interesting one I've been thinking about...I suppose if I start using that feature that any users working with an older version of Kube however the example will not work so I'm kind of torn between using it or waiting till some of these features get more out there:  initcontainers, dynamic provisioning, StatefulSet....are these examples meant to be leading edge with respect to kube features or merely examples of using Helm?

**Comment by jmccormick2001** [comment]
question...I see the postgres example refers to a PVC storage class of "storageClass: generic"

I can't seem to find where that storage class is documented or how to enable it in the kube ...any pointers?  is this an experimental feature?

**Comment by viglesiasce** [comment]
Hey @jmccormick2001!

We are in a bit of a transitional state with PVC's in that they have beta support but we are using the alpha annotation to enable them. With the alpha annotation the value is ignored and is only checked to be set to something. 

For more info on the path forward check out #116

**Comment by lubo** [comment]
@jmccormick2001 I see you've been working on https://github.com/CrunchyData/postgres-operator/tree/master/chart/postgres-operator. Why did you close this PR, though? Are you no longer willing to put the afford into getting the chart merged into this repository?

**Comment by jmccormick2001** [comment]
I might give this another shot.

**Pull request closed without merging**